src/core/providers/gdal/qgsgdalprovider.cpp

```diff
@@ -3268,6 +3268,34 @@ bool QgsGdalProvider::initIfNeeded()
     return false;
   }
 
+#if GDAL_VERSION_NUM >= GDAL_COMPUTE_VERSION(3,8,0)
+  // Set INTERPOLATE option for VR BAG
+  if ( QString( GDALGetDriverShortName( GDALGetDatasetDriver( mGdalBaseDataset ) ) ) == QLatin1String( "BAG" ) && QString{ GDALGetMetadataItem( mGdalBaseDataset, "HAS_SUPERGRIDS", nullptr ) } == QLatin1String( "TRUE" ) )
+  {
+    QVariantMap parts = decodeGdalUri( gdalUri );
+    QStringList openOptions = parts.value( QStringLiteral( "openOptions" ) ).toStringList();
+    bool hasModeOption = false;
+
+    for ( const QString &option : std::as_const( openOptions ) )
+    {
+      if ( option.startsWith( QStringLiteral( "MODE=" ) ) )
+      {
+        hasModeOption = true;
+        break;
+      }
+    }
+
+    if ( ! hasModeOption )
+    {
+      openOptions.append( QStringLiteral( "MODE=INTERPOLATED" ) );
```

Comment:
if a user opens the "Add raster layer" dialog and selects a BAG dataset, with that make the MODE option combo box select INTERPOLATED ?

Reply:
No, I thought about that but I thought it was better to leave the <DEFAULT>.

What I am changing here is the default, which is INTERPOLATED if it is a VR-BAG.

This needs to happen in the provider because the layer may be added bypassing the data source dialog (e.g. with a drag and drop from the browser), so I implemented the logic at the provider level that makes this the default option for VR-BAGs.

Actually I am wondering if we should make MODE=INTERPOLATED the default for VR-BAG in GDAL itself.

Or, I could implement a logic in the data source select to auto-select ITERPOLATED for VR-BAGs, I'd just need to open the dataset to get the metadata item for supergrids.

Reply:
> Actually I am wondering if we should make MODE=INTERPOLATED the default for VR-BAG in GDAL itself.

I thought about that, as that would make QGIS life simpler indeed, but I'm slightly uncomfortable as this is "creating" information rather than just exposing actual file content.

Reply:
> > Actually I am wondering if we should make MODE=INTERPOLATED the default for VR-BAG in GDAL itself.
> 
> I thought about that, as that would make QGIS life simpler indeed, but I'm slightly uncomfortable as this is "creating" information rather than just exposing actual file content.

Right, would you recommend to remove the logic from the provider and move it to the widget? 

This way it would use the default mode when opening with drag&drop from the browser though, which is probably not what a user desire.

Something to consider is that this VR-BAG is probably a very uncommon use case.

Reply:
> Right, would you recommend to remove the logic from the provider and move it to the widget?

not necessarily. I'd say what you implemented is good enough. The cherry on the cake would be that the data source would  auto-select INTERPOLATED for VR-BAGs to make it clear to the user what is going on.

Reply:
> > Right, would you recommend to remove the logic from the provider and move it to the widget?
> 
> not necessarily. I'd say what you implemented is good enough. The cherry on the cake would be that the data source would auto-select INTERPOLATED for VR-BAGs to make it clear to the user what is going on.

Ok, I think I can do that.

Reply:
done